auth-server/src/main/java/com/antonio/authserver/service/UserService.java

```diff
@@ -99,43 +98,49 @@ public AppUserDto update(AppUserDto appUserDto) {
         return appUserDto;
     }
 
-    public void updateUserByUsername(String realmName,String username, AppUserDto appUserDto) {
+    public void updateUserByUsername(String realmName, String username, AppUserDto appUserDto) {
         appUserDto.setUsername(appUserDto.getUsername().replaceAll("\\s+", ""));
-        AppUser appUser = appUserRepository.findByUsernameAndRealmName(username,realmName).orElseThrow(() -> new CustomException(
+        AppUser appUser = appUserRepository.findByUsernameAndRealmName(username, realmName).orElseThrow(() -> new CustomException(
                 "User with the username [ " + username + " ] could not be found!", HttpStatus.NOT_FOUND));
         if (appUserDto.getUsername().equals(""))
             throw new CustomException("The inserted User cannot be null!", HttpStatus.BAD_REQUEST);
         appUser.setUsername(appUserDto.getUsername());
         appUser.setPassword(appUserDto.getPassword());
-        if(appUserDto.getToken() != null) {
-            appUser.setToken(appUserDto.getToken());
-        }
-        if(appUserDto.getRefreshToken() != null){
-            appUser.setRefreshToken(appUserDto.getRefreshToken());
-        }
+        appUser.setToken(appUserDto.getToken());
+        appUser.setRefreshToken(appUserDto.getRefreshToken());
+
 
         appUserRepository.save(appUser);
     }
 
-    public AppUser addRole(String realmName,String username, Role role) throws CustomException {
-        AppUser appUser = appUserRepository.findByUsernameAndRealmName(username,realmName).orElseThrow(() -> new CustomException(
+    public List<AppUserDto> getUsersWithoutAdmins(String realmName) {
+        Role amdinRole = roleRepository.findByName("ROLE_ADMIN").orElseThrow(() -> new CustomException("Role not found", HttpStatus.BAD_REQUEST));
+        List<AppUser> users = appUserRepository.findAll()
+                .stream()
+                .filter(u -> !u.getRoles().contains(amdinRole))
```

Comment:
nu mai fa read pentru admin, ci cand faci filter, fa direct dupa role name "ROLE_ADMIN"

Reply:
Pai in filter am nevoie de obiect, nu pot pune "contains" dupa nume